fe/fe-core/src/main/java/org/apache/doris/analysis/SelectStmt.java

```diff
@@ -1263,6 +1269,9 @@ private void analyzeAggregation(Analyzer analyzer) throws AnalysisException {
                 havingClauseAfterAnaylzed = havingClause.substitute(aliasSMap, analyzer, false);
             }
             havingClauseAfterAnaylzed = rewriteQueryExprByMvColumnExpr(havingClauseAfterAnaylzed, analyzer);
+            if (havingClauseAfterAnaylzed.getType().isTinyint()) {
```

Comment:
i think it should be `!isBoolean()`

Reply:
yes, in master branch, old planner should be the same behavior as new planner